### ISSUE TITLE
v1.96 (Windows): Error mining etchash&rvn on NVidia 1660 Super 6GB

### ISSUE THREAD

**Issue by UselessGuru** [sp-hash/TeamBlackMiner#380]
`Bin\TeamBlackMiner-v1.96\TBMiner.exe --algo etc+rvn --hostname etchash.auto.nicehash.com --wallet #######################################.################--port 9200 --server-passwd x --rvn-hostname kawpow.mine.zergpool.com --rvn-wallet ***************************** --rvn-port 3638 --api --api-version 1.4 --api-port 4006 --cuda-devices [0]`

00:00:00 [2023-08-24 16:16:25.734]
00:00:00 [2023-08-24 16:16:25.802] OpenCL driver version: 536.99
00:00:00 [2023-08-24 16:16:25.828] Cuda driver version: 12.2
00:00:00 [2023-08-24 16:16:25.828] Cuda runtime API version: 12.0
00:00:00 [2023-08-24 16:16:25.828]
00:00:00 [2023-08-24 16:16:25.828]
00:00:00 [2023-08-24 16:16:25.828]
00:00:00 [2023-08-24 16:16:25.828] Welcome to Team Black Miner 1.96
00:00:00 [2023-08-24 16:16:25.828]
00:00:00 [2023-08-24 16:16:25.828]
00:00:00 [2023-08-24 16:16:25.942] GPU0 NVIDIA GeForce GTX 1660 SUPER 6GB added to miner as Cuda device
00:00:00 [2023-08-24 16:16:25.944] Trying kawpow.mine.zergpool.com:3638 (kawpow)
00:00:00 [2023-08-24 16:16:25.944] Trying etchash.auto.nicehash.com:9200 (etchash)
00:00:00 [2023-08-24 16:16:25.944] Api server started at 127.0.0.1:4006
00:00:00 [2023-08-24 16:16:26.005] etchash.auto.nicehash.com: Pool responded successfully to subscribe
00:00:00 [2023-08-24 16:16:26.056] etchash.auto.nicehash.com: Pool responded successfully to extranonce subscribe
00:00:00 [2023-08-24 16:16:26.056] GPU0 Started Cuda work thread
00:00:00 [2023-08-24 16:16:26.056] GPU0 XIntensity set to 1024
00:00:00 [2023-08-24 16:16:26.088] etchash.auto.nicehash.com: Pool responded successfully to authorize
00:00:00 [2023-08-24 16:16:26.103] etchash.auto.nicehash.com: Difficulty is set to 0.396590
00:00:00 [2023-08-24 16:16:28.303] GPU0 Starting generating etchash dag (Epoch: 303)
00:00:00 [2023-08-24 16:16:28.395] GPU0 DAG generated in 2.26 seconds (3530750kb)
00:00:00 [2023-08-24 16:16:28.395] GPU0 kernel 1 selected
00:00:00 [2023-08-24 16:16:28.395] Job 00000000d422320d
00:00:00 [2023-08-24 16:16:29.080] kawpow.mine.zergpool.com: Pool responded to subscribe
00:00:00 [2023-08-24 16:16:29.149] kawpow.mine.zergpool.com: Pool responded successfully to extranonce subscribe
00:00:00 [2023-08-24 16:16:29.227] kawpow.mine.zergpool.com: Pool responded successfully to authorize
00:00:00 [2023-08-24 16:16:29.274] GPU0 Starting generating kawpow dag (Epoch: 88)
00:00:00 [2023-08-24 16:16:35.781] GPU0 DAG generated in 6.51 seconds (1769470kb)
00:00:00 [2023-08-24 16:16:35.782] Job b303
00:00:00 [2023-08-24 16:16:43.050] **C:\code\tbminer\source\sp_kawpow\cuda_kawpow.cpp:cuda_kawpow_init_epoch():122: A cuda runtime error occured: invalid argument**
00:00:00 [2023-08-24 16:16:43.050] Shutting down threads
00:00:00 [2023-08-24 16:16:43.051] Stratum thread exited
00:00:00 [2023-08-24 16:16:43.051] Stratum thread exited
00:00:00 [2023-08-24 16:16:43.051] Control thread exited
00:00:00 [2023-08-24 16:16:43.054] GPU0 thread exited

Is the miner running out of memory? If so a more explanatory error message would be helpful ;-)

**Comment by sp-hash** [comment]
ETC epoc 302  3.367 GB
RVN epoc  392  4.063 GB

So mining ETC+RVN will require 7.4 GB free on the gpu

**Comment by UselessGuru** [comment]
Not quite. 

00:00:00 [2023-08-24 16:16:29.274] GPU0 Starting generating kawpow dag (Epoch: 88)
00:00:00 [2023-08-24 16:16:35.781] GPU0 DAG generated in 6.51 seconds (1769470kb)

00:00:00 [2023-08-24 16:16:28.303] GPU0 Starting generating etchash dag (Epoch: 303)
00:00:00 [2023-08-24 16:16:28.395] GPU0 DAG generated in 2.26 seconds (3530750kb)

That's about 5.3GB. If memory is not enough then a descriptive error message would help.

**Comment by UselessGuru** [comment]
Any news?

**Comment by sp-hash** [comment]
works on linux

**Comment by UselessGuru** [comment]
>> works on linux
That does not help me in any way.
If it does NOT work under windows, then please document so. Thank you!

**Comment by UselessGuru** [comment]
Still the same error with 2.01:

`TBMiner.exe --algo eth+vtc --verthash-data ..\..\Cache\VertHash.dat --hostname europe.ethash-hub.miningpoolhub.com --wallet ***************.****************--port 20565 --server-passwd x --vtc-hostname stratum1.hashcryptos.com --vtc-wallet ******************************.************* --vtc-passwd *************** --vtc-port 9991 --api --api-version 1.4 --api-port 4005 --cl-devices [1]`

```
00:00:00 [2023-09-30 18:46:31.420]
00:00:00 [2023-09-30 18:46:31.498] OpenCL driver version: 536.99
00:00:00 [2023-09-30 18:46:31.515] Cuda driver version: 12.2
00:00:00 [2023-09-30 18:46:31.515] Cuda runtime API version: 12.0
00:00:00 [2023-09-30 18:46:31.515]
00:00:00 [2023-09-30 18:46:31.515]
00:00:00 [2023-09-30 18:46:31.515]
00:00:00 [2023-09-30 18:46:31.515] Welcome to Team Black Miner 2.01
00:00:00 [2023-09-30 18:46:31.515]
00:00:00 [2023-09-30 18:46:31.515]
00:00:00 [2023-09-30 18:46:31.515] GPU0 Ellesmere 8GB added to miner as OpenCL device
00:00:00 [2023-09-30 18:46:31.515] Trying to read verthash data file from disk
00:00:00 [2023-09-30 18:46:31.515] Api server started at 127.0.0.1:4005
00:00:00 [2023-09-30 18:46:32.008] Verthash data file has been loaded succesfully
00:00:00 [2023-09-30 18:46:38.297] Verthash data file has been verified succesfully
00:00:00 [2023-09-30 18:46:38.297] Trying stratum1.hashcryptos.com:9991 (verthash)
00:00:00 [2023-09-30 18:46:38.297] Trying europe.ethash-hub.miningpoolhub.com:20565 (ethash)
00:00:00 [2023-09-30 18:46:38.313] GPU0 Started CL work thread
00:00:00 [2023-09-30 18:46:38.313] GPU0 Intensity set to 256
00:00:00 [2023-09-30 18:46:38.361] stratum1.hashcryptos.com: Pool responded successfully to subscribe
00:00:00 [2023-09-30 18:46:38.408] stratum1.hashcryptos.com: Pool responded successfully to extranonce subscribe
00:00:00 [2023-09-30 18:46:38.440] stratum1.hashcryptos.com: Pool responded successfully to authorize
00:00:00 [2023-09-30 18:46:38.456] stratum1.hashcryptos.com: Difficulty is set to 0.500000
00:00:00 [2023-09-30 18:46:38.488] europe.ethash-hub.miningpoolhub.com: Pool responded successfully to subscribe
00:00:00 [2023-09-30 18:46:38.600] europe.ethash-hub.miningpoolhub.com: Difficulty is set to 3.000000
00:00:00 [2023-09-30 18:46:38.600] GPU0 Load verthash data size 1283457024
00:00:00 [2023-09-30 18:46:38.631] GPU0 Starting generating zil dag (Epoch: 1)
00:00:00 [2023-09-30 18:46:38.710] europe.ethash-hub.miningpoolhub.com: Pool responded successfully to authorize
00:00:00 [2023-09-30 18:46:39.498] GPU0 Verthash data loaded to gpu
00:00:00 [2023-09-30 18:46:39.503] Job 76657274636f696e2d313737-205  (verthash)
00:00:00 [2023-09-30 18:46:46.314] GPU0 DAG generated in 7.70 seconds (1056767kb)
00:00:00 [2023-09-30 18:46:47.498] GPU0 Starting generating ethash dag (Epoch: 275)
00:00:00 [2023-09-30 18:46:58.435] VERTHASH: GPU0 Accepted (1/1/0) Reply: 565ms
00:00:00 [2023-09-30 18:47:05.125] VERTHASH: GPU0 Accepted (2/2/0) Reply: 47ms
00:00:00 [2023-09-30 18:47:07.086] GPU0 DAG generated in 20.73 seconds (3301375kb)

```
Then back to command line.
1.056 + 3.30 GB should easily fit in 8GB.

**Comment by sp-hash** [comment]
But you are mining ETH+VTC in the bat file and ethereum pow need 6GB memory for the pow. ZIL 1.3GB, ZIL 1,1GB, ETH 5.9 GB

So change to ETC+VTC port and config

**Comment by UselessGuru** [comment]
> But you are mining ETH+VTC in the bat file and ethereum **pow need 6GB memory for the pow**

No, Ethash **DAG only needs 3.3 GB for this coin**. So both DAGs should easily fint in 8GB, 6GB even.
If there really IS a memory issue, than a meaningful error message would help tremendously.

00:00:00 [2023-09-30 18:46:46.314] GPU0 DAG generated in 7.70 seconds (1056767kb)
00:00:00 [2023-09-30 18:46:47.498] GPU0 Starting generating ethash dag (Epoch: 275)
00:00:00 [2023-09-30 18:46:58.435] VERTHASH: GPU0 Accepted (1/1/0) Reply: 565ms
00:00:00 [2023-09-30 18:47:05.125] VERTHASH: GPU0 Accepted (2/2/0) Reply: 47ms
00:00:00 [2023-09-30 18:47:07.086] GPU0 DAG generated in 20.73 seconds (3301375kb)

**Comment by UselessGuru** [comment]
Fixed in 2.02
